test/test_events_app.py

```diff
@@ -31,48 +32,114 @@ def setUp(self):
         with self.app.app_context():
             # create all tables
             db.create_all()
+            
+    def register_user(self):
+        """This helper method helps register a test user."""
+        user_data = {
+            'first_name': 'new',
+            'last_name': 'user',
+            'email': 'frank@test.com',
+            'password': 'test_password'
+        }
+        return self.client().post('/api/auth/register/', data=user_data)
 
+    def login_user(self):
+        """This helper method helps log in a test user."""
+        user_data = {
+            'email': 'frank@test.com',
+            'password': 'test_password'
+        }
+        return self.client().post('/api/auth/login/', data=user_data)
 
     def test_create_event(self):
         """Test API can create an event (POST request)"""
-        resp = self.client().post('/api/events/', data=self.new_event)
+        #register a user then log them in
+        self.register_user()
+        result = self.login_user()
+        #obtain the access token
+        access_token = json.loads(result.data.decode())['access_token']
+
+        resp = self.client().post(
+            '/api/events/',
+            headers=dict(Authorization="Bearer " + access_token),
+            data=self.new_event)
         self.assertEqual(resp.status_code, 201)
         self.assertIn('Barbecue party', str(resp.data))
 
     def test_retrieve_all_events(self):
         """Test API can retrieve all events (GET request)."""
-        resp = self.client().post('/api/events/', data=self.new_event)
+        self.register_user()
+        result = self.login_user()
+        access_token = json.loads(result.data.decode())['access_token']
```

Comment:
you can move this process of getting the access token into a method and call that one method instead of the 3 step in every test

Reply:
good idea, ill refactor and take that into consideration